src/lib/renderElement.js

```diff
@@ -21,42 +31,143 @@ function updateAttributes() {
   //     - TODO: 'on'으로 시작하는 속성을 이벤트 리스너로 처리
   //     - 주의: 직접 addEventListener를 사용하지 않고, eventManager의 addEvent와 removeEvent 함수를 사용하세요.
   //     - 이는 이벤트 위임을 통해 효율적으로 이벤트를 관리하기 위함입니다.
+
+  newProps = newProps && typeof newProps === "object" ? newProps : {};
+  oldProps = oldProps && typeof oldProps === "object" ? oldProps : {};
```

Comment:
```suggestion
  newProps = newProps || {};
  oldProps = oldProps || {};
```

여기서는 이렇게 객체가 아니더라도 falsy한 값이면 빈 객체로 처리되도록 축약할 수 있을 것 같아요! 😆

Reply:
확인 감사해요:) !!